plugins/inputs/kibana/kibana.go

```diff
@@ -187,15 +199,27 @@ func (k *Kibana) gatherKibanaStatus(baseUrl string, acc telegraf.Accumulator) er
 	tags["status"] = kibanaStatus.Status.Overall.State
 
 	fields["status_code"] = mapHealthStatusToCode(kibanaStatus.Status.Overall.State)
-
-	fields["uptime_ms"] = kibanaStatus.Metrics.UptimeInMillis
 	fields["concurrent_connections"] = kibanaStatus.Metrics.ConcurrentConnections
-	fields["heap_max_bytes"] = kibanaStatus.Metrics.Process.Mem.HeapMaxInBytes
-	fields["heap_used_bytes"] = kibanaStatus.Metrics.Process.Mem.HeapUsedInBytes
 	fields["response_time_avg_ms"] = kibanaStatus.Metrics.ResponseTimes.AvgInMillis
 	fields["response_time_max_ms"] = kibanaStatus.Metrics.ResponseTimes.MaxInMillis
 	fields["requests_per_sec"] = float64(kibanaStatus.Metrics.Requests.Total) / float64(kibanaStatus.Metrics.CollectionIntervalInMilles) * 1000
 
+	Version, err := strconv.ParseFloat(strings.Join(strings.Split(kibanaStatus.Version.Number, ".")[:2], "."), 64)
+	if err != nil {
+		return err
+	}
+
+	if Version >= 6.5 {
+		fields["uptime_ms"] = kibanaStatus.Metrics.Process.UptimeInMillis
+		fields["heap_total_bytes"] = kibanaStatus.Metrics.Process.Memory.Heap.TotalInBytes
```

Comment:
For backwards compatibility, either record this only as only `heap_max_bytes` or as both `heap_total_bytes` and `heap_max_bytes`.  Make sure to add a comment so we can remember why we did this.

Reply:
code updated along with the comments as per the suggestions